astropy/visualization/mpl_normalize.py

```diff
@@ -212,6 +212,231 @@ def inverse(self, values, invalid=None):
+        ``percent``.
+
+    max_percent : float, optional
+        The percentile value used to determine the pixel value of
+        maximum cut level. The default is 100.0. ``max_percent``
+        overrides ``percent``.
+
+    vmin : float, optional
+        The pixel value of the minimum cut level. Data values less
+        than ``vmin`` will set to ``vmin`` before stretching the
+        image. The default is the image minimum. ``vmin`` overrides
+        ``min_percent``.
+
+    vmax : float, optional
+        The pixel value of the maximum cut level. Data values greater
+        than ``vmax`` will set to ``vmax`` before stretching the
+        image. The default is the image maximum. ``vmax`` overrides
+        ``max_percent``.
+
+    power : float, optional
+        The power index for ``stretch='power'``. The default is 1.0.
+
+    log_a : float, optional
+        The log index for ``stretch='log'``. The default is 1000.
+
+    asinh_a : float, optional
+        For ``stretch='asinh'``, the value where the asinh curve
+        transitions from linear to logarithmic behavior, expressed as a
+        fraction of the normalized image. Must be in the range between 0
+        and 1. The default is 0.1.
+
+    sinh_a : float, optional
+        The scaling parameter for ``stretch='sinh'``. The default is
+        0.3.
+
+    clip : bool, optional
+        If `True`, data values outside the [0:1] range are clipped to
+        the [0:1] range.
+
+    invalid : None or float, optional
+        Value to assign NaN values generated by the normalization. NaNs
+        in the input ``data`` array are not changed. For matplotlib
+        normalization, the ``invalid`` value should map to the
+        matplotlib colormap "under" value (i.e., any finite value < 0).
+        If `None`, then NaN values are not replaced. This keyword has no
+        effect if ``clip=True``.
+
+    See Also
+    --------
+    simple_norm
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+        from astropy.visualization import SimpleNorm
+
+        image = np.arange(65536).reshape((256, 256))
+        snorm = SimpleNorm('sqrt', percent=98)
+        norm = snorm(image)
+        fig, ax = plt.subplots()
+        axim = ax.imshow(image, norm=norm, origin='lower')
+        fig.colorbar(axim)
+    """
+
+    def __init__(
+        self,
+        stretch="linear",
+        percent=None,
+        min_percent=None,
+        max_percent=None,
+        vmin=None,
+        vmax=None,
+        power=1.0,
+        log_a=1000,
+        asinh_a=0.1,
+        sinh_a=0.3,
+        clip=False,
+        invalid=-1.0,
+    ):
+        if percent is not None:
+            interval = PercentileInterval(percent)
+        elif min_percent is not None or max_percent is not None:
+            interval = AsymmetricPercentileInterval(
+                min_percent or 0.0, max_percent or 100.0
+            )
```

Comment:
Hum, `max_percent or 100.0` will silently replace an explicit `max_percent=0.0` with `100.0`, which I think would be surprising and could lead to subtle and hard to debug situations.
Is there a good reason for `AsymmetricPercentileInterval.__init__` not to have default values for `min_percent` and `max_percent` ?
How about the following

```suggestion
            interval = AsymmetricPercentileInterval(
                lower_percentile=min_percent, upper_percentile=max_percent
            )
```

together with this patch:
```patch
diff --git a/astropy/visualization/interval.py b/astropy/visualization/interval.py
index 66fac6a117..b370da0152 100644
--- a/astropy/visualization/interval.py
+++ b/astropy/visualization/interval.py
@@ -156,9 +156,9 @@ class AsymmetricPercentileInterval(BaseInterval):
         sampled from the array (with replacement).
     """

-    def __init__(self, lower_percentile, upper_percentile, n_samples=None):
-        self.lower_percentile = lower_percentile
-        self.upper_percentile = upper_percentile
+    def __init__(self, lower_percentile=None, upper_percentile=None, n_samples=None):
+        self.lower_percentile = lower_percentile if lower_percentile is not None else 0.0
+        self.upper_percentile = upper_percentile if upper_percentile is not None else 100.0
         self.n_samples = n_samples

     def get_limits(self, values):
```

Reply:
Good catch!  This is fixed now!